x-pack/platform/plugins/shared/onechat/server/services/agents/modes/default/run_chat_agent.ts

```diff
@@ -69,6 +69,9 @@ export const runDefaultAgentMode: RunChatAgentFn = async (
     sendEvent: eventEmitter,
   });
 
+  const cycleLimit = 10;
+  const graphRecursionLimit = cycleLimit * 2 + 8; // 2 steps per cycle + 3 additional steps + some extra buffer
```

Comment:
can we have more comments around this   i thought recursion limit was just tools calls ... but reading your pr description it's not 1-to-1. I don't fully get this math
```
2 steps per cycle + 3 additional steps + some extra buffer
```
trying to keep bus factor down :D